README.md

```diff
@@ -1,103 +1,124 @@
+The tested ideal environment for building this module is Debian Bullseye (used in our CI).
 
-## Deploy to robot
+You can also use Ubuntu 22.04 for testing, but the resulting C++ builds will not be binary compatible with Bullseye.
 
-These instructions assume you're targeting something similar to Debian bullseye.
 
-First install libraries on your target system:
+To setup for building this module:
+
+1. Install the apt packages in [apt-setup.sh](apt-setup.sh).
+2. Install CMake. Ensure your version of CMake matches the [CMAKE_VERSION in this repository's github workflow](https://github.com/viamrobotics/module-example-cpp/blob/af057c9169a730b2c5de5c26f877c0419fb52c19/.github/workflows/build.yml#L7).
+	- To install on Ubuntu 22.04, run `snap install cmake` or follow [CMake's download instructions](https://cmake.org/download/).
+3. [Install rust](https://www.rust-lang.org/tools/install) (needed for `viam_rust_utils`).
+
+## Build and Run
+
+These instructions assume a target system similar to Debian Bullseye.
+First, install the required libraries on your target system:
 
 ```sh
 sudo apt install -qqy libgrpc++1 libboost-log1.74.0
 ```
 
-Then set this JSON config for your robot in the app.viam.com webapp.
+Then follow the C++ instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `wifi:cpp` model from the `module-example-cpp` module.
+
+
+## Configure your Example C++ Module
 
-Go to 'Config' tab, then change the 'Mode:' selector from 'Builder' to 'Raw JSON', then paste in the following.
+> [!NOTE]
+> Before configuring your example module, you must [create a robot](https://docs.viam.com/manage/fleet/robots/#add-a-new-robot).
 
-```json
+Navigate to the **Config** tab of your robot’s page in [the Viam app](https://app.viam.com/).
+Click on the **Components** subtab and click **Create component**.
+Select **Sensor**, then search for the `wifi:cpp` model. Give your resource a name of your choice, click **Create** and then click **Save config**.
+
+Next, change the `Mode:` selector from `Builder` to `Raw JSON`, and verify that `"components"` array now includes the `viam:wifi:cpp` model and that `"modules"` array includes `viam:module-example-cpp` module. 
+
+Your `Raw JSON` configuration should now include both: 
+
+```json {class="line-numbers linkable-line-numbers"}
 {
   "components": [{
-	"attributes": {},
-	"model": "viam:wifi:cpp",
-	"depends_on": [],
-	"name": "wifi-cpp",
-	"type": "generic"
+    "attributes": {},
+		"depends_on": [],
+    "name": "wifi-cpp",
+		"model": "viam:wifi:cpp",
+    "type": "sensor",
+		"namespace": "rdk"
   }],
   "modules": [{
-	"module_id": "viam:module-example-cpp",
-	"type": "registry",
-	"name": "cpp",
-	"version": "0.0.0-main.15"
+		"type": "registry",
+	  "name": "viam_module-example-cpp",
+    "module_id": "viam:module-example-cpp",
+    "version": "0.0.3"
```

Comment:
the indenting here looks a little funky, but I don't work with json much! Is this intentional?

Reply:
Agreed, let's try to match the indentation in the App; this is what I get:

```
{
  "modules": [
    {
      "type": "registry",
      "name": "viam_module-example-cpp",
      "module_id": "viam:module-example-cpp",
      "version": "0.0.3"
    }
  ],
  "components": [
    {
      "name": "my-wifi-cpp",
      "model": "viam:wifi:cpp",
      "type": "sensor",
      "namespace": "rdk",
      "attributes": {},
      "depends_on": []
    }
  ]
}
```

Reply:
also in case this is in here because it was in my old instructions -- we don't need to edit json for registry modules anymore, the UX takes care of it